securedrop/secure_tempfile.py

```diff
@@ -1,12 +1,16 @@
 import base64
 import os
 from tempfile import _TemporaryFileWrapper
+# python-gnupg will not recognize our SecureTemporaryFile as a stream-like type
+# and will attempt to call encode on it, thinking it's a string-like type. To
+# avoid this we add it the list of stream-like types.
+from gnupg._util import _STREAMLIKE_TYPES
+_STREAMLIKE_TYPES.append(_TemporaryFileWrapper)
```

Comment:
It's kind of a hack that we use an internal type (`_TemporaryFileWrapper`, prepended by a `_`), and we've already had at least one person complain about it: https://github.com/freedomofpress/securedrop/issues/888. I'd prefer to minimize our use if possible. Also, appending `SecureTemporaryFile` instead of `_TemporaryFileWrapper` is more explicit and that is almost always better.